app/models/miq_report/search.rb

```diff
@@ -99,6 +100,13 @@ def paged_view_search(options = {})
     else
       targets = db_class
     end
+
+    unless selected_ids.nil?
+      targets = targets.select do |one_target|
```

Comment:
Looks much better, I was too complicated, was thinking, what if `selected_ids` will have IDs which are not present in `target` and over-complicated it. But I'll use the part you suggested.